hie-base/Haskell/Ide/Engine/PluginTypes.hs

```diff
@@ -1,3 +1,7 @@
+{-# LANGUAGE InstanceSigs #-}
+{-# LANGUAGE UndecidableInstances #-}
```

Comment:
I think Singletons are fine, it is more about limiting the scope of the
UndecidableInstances.

Which the people on #ghc tell me are not that bad, but if enabled you can
potentially do stupid things, as you turn off certain checks





On Thu, Jan 7, 2016 at 6:23 PM, Moritz Kiefer <notifications@github.com>
wrote:

> In hie-base/Haskell/Ide/Engine/PluginTypes.hs
> <https://github.com/haskell/haskell-ide-engine/pull/152#discussion_r49091487>
> :
>
> > @@ -1,3 +1,7 @@
> > +{-# LANGUAGE InstanceSigs #-}
> > +{-# LANGUAGE UndecidableInstances #-}
>
> Dropping the dependency on singletons or moving the definitions into a
> separate module?
> The latter should be easy. Dropping the dependency on singletons shouldn’t
> be too hard. The disadvantage is that it introduces some ugly duplication
> since the definitions for the singletons and the normal data types need to
> match. Maybe there is also a way to only import a minimal part of
> singletons which doesn’t require UndecidableInstances.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/haskell/haskell-ide-engine/pull/152/files#r49091487>.
>